src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -65,20 +68,31 @@ def __init__(
   def _combined_invalidating_fileset_from_globs(glob_strs, root):
     return set.union(*(Fileset.globs(glob_str, root=root)() for glob_str in glob_strs))
 
+  def _get_snapshot(self, glob):
+    print(dir(self._scheduler))
+    """Returns a Snapshot of the input glob"""
+    return self._scheduler_session.product_request(Snapshot, subjects=[PathGlobs(glob)])[0]
+
+  def _get_files(self, snapshots):
+    """Returns a list of the files corresponding to the list of input snapshots"""
+    return [snapshot.files for snapshot in snapshots][0]
+
   def setup(self, services):
     """Service setup."""
     super(SchedulerService, self).setup(services)
     # Register filesystem event handlers on an FSEventService instance.
     self._fs_event_service.register_all_files_handler(self._enqueue_fs_event)
 
-    # N.B. We compute this combined set eagerly at launch with an assumption that files
+    # N.B. We compute the invalidating fileset eagerly at launch with an assumption that files
     # that exist at startup are the only ones that can affect the running daemon.
     if self._invalidation_globs:
-      self._invalidating_files = self._combined_invalidating_fileset_from_globs(
+      invalidating_files = self._combined_invalidating_fileset_from_globs(
```

Comment:
The `_combined_invalidating_fileset_from_globs` method can be completely removed, and `self._invalidation_globs` can be passed directly to `self._get_snapshot`.